cmd/carbonapi/main.go

```diff
@@ -496,6 +504,10 @@ func setUpConfig(logger *zap.Logger) {
 		graphite.Register(fmt.Sprintf("%s.request_cache_misses", pattern), apiMetrics.RequestCacheMisses)
 		graphite.Register(fmt.Sprintf("%s.request_cache_overhead_ns", pattern), apiMetrics.RenderCacheOverheadNS)
 
+		for i := 0; i <= config.Buckets; i++ {
+			graphite.Register(fmt.Sprintf("%s.requests_in_%dms_to_%dms", pattern, i*100, (i+1)*100), bucketEntry(i))
```

Comment:
There was an old Feature-request for carbonzipper and carbonapi to call last bucket as `requests_in_Xms_to_inf`. Maybe as it's a good idea to implement that as part of this PR, otherwise we'll end up having `(i+1)*100` for last bucket forever.

Reply:
See:
https://godoc.org/go.uber.org/zap#SamplingConfig
https://godoc.org/go.uber.org/zap#AtomicLevel.ServeHTTP

and docs around that.

Reply:
We thought about it and decided we'd like the last bucket to also be `requests_in_Xms_to_(X+100)ms`, because if we don't and change our minds at some point about how many buckets we want to have (say we'd like to have 20 instead of 10), then we'll have two `_to_inf` buckets lying around in Graphite.

Reply:
But in that case the first "to_inf" bucket will contain only nulls after that change. Otherwise it'll continue to exist, but the meaning of the values will suddenly change.

I understand that it's actually choosing lesser of two evils, so up to you to make decision on default behavior. If you think it's ok, feel free to merge PR.

I'd still suggest to at least add an option in future that will allow to change that behavior.

Reply:
Both make sense. I vote we defer this to [an issue](https://github.com/go-graphite/carbonapi/issues/350) so we can get the non-infinite-loop fix in here. For now, the situation will at least be no worse than in carbonzipper.

Reply:
Agree.